Assets/Scripts/Weapons/BehaviourTypes/BulletMono.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BulletMono : MonoBehaviour
+{
+    float _damage;
+    public float Damage 
+    { 
+        get => _damage; 
+        set => _damage = value; 
+    }
+
+    float _bulletSpeed = 1f;
+    public float BulletSpeed {
+        get => _bulletSpeed;
+        set => _bulletSpeed = value;
+    }
+
+    float _bulletSpawnTime;
+
+    public static void create(Vector3 position, Vector3 direction, Dictionary<string, float> statsModifiers)
+    {
+        GameObject newObject = Instantiate(Resources.Load("Bullet")) as GameObject;
+        newObject.transform.position = position;
+        newObject.transform.forward = direction;
+
+        BulletMono bulletComponent = newObject.GetComponent<BulletMono>();
+        bulletComponent.Damage = statsModifiers["damage"];
+        bulletComponent.BulletSpeed = statsModifiers["bulletSpeed"];
+    }
+
+    void Start()
+    {
+        _bulletSpawnTime = Time.time;
+    }
+
+    void Update()
+    {
+        transform.position += transform.forward * _bulletSpeed * Time.deltaTime; 
+        if (Time.time - _bulletSpawnTime > 10f)
+        {
+            Destroy(gameObject);
+        }
+    }
+
+    void OnTriggerEnter(Collider c){
+        switch (c.gameObject.tag)
```

Comment:
Bullets should be general- spawnable by both the player and enemies. Currently, this implementation assumes the player is the owner of the bullet.

Reply:
The idea we had during the Game Jam was to have the bullet check for a Stats component on the hit game object, then affect the game object based on the stats. We might want to go this route again. Allows for some decoupling.

Reply:
https://github.com/MaXeraph/Adrenarun/pull/4/commits/0b5cd36f570d399d82d5f06f04b335613603e023#diff-c265438d87075a9b2ee6d9ed93d8eca0d327a8316806ad19dae6193a508e35d2R34-R41